MovingBlocks/Terasology
# Add GUI screen to name a recording

### Contains
Adds the screen `nameRecordingScreen` after the selection of a game to record, prompting the player to make a unique name for their new recording. The screen prevents naming a recording with the same name as an existing recording, resolving #3465. If the player enters an invalid file name (blank, entirely invalid characters, existing recording), the player is prompted to re-enter a name.

The structure of `RecordScreen` has been changed, now primary file operations and loading for the recording are now handled in `NameRecordingScreen`.

### How to test

- Create a new recording, and set the name to whatever you want!
- Erronous recording names should be handled as follows:
   - If the text field is left blank, the file name should be reported invalid.
   - If the recording name is made of entirely invalid characters, the file name should be reported invalid.
   - If the recording name already exists, the file name should be reported as a duplicate.
   - If the recording name has some invalid characters, the manifest will have the full name while the directory name will have these characters removed.

### Translation

- Four new translation strings have been added for `nameRecordingScreen`
- I have added an English and French translation for all four strings.
- I'm not a native French speaker, so if somebody can confirm/fix my French translation that would be greatly appreciated! _(it has been a long time since french class.....)_

@iaronaraujo
